i6_models/config.py

```diff
@@ -51,3 +54,25 @@ def _validate_types(self) -> None:
 
     def __post_init__(self) -> None:
         self._validate_types()
+
+
+ConfigType = TypeVar("ConfigType", bound=ModelConfiguration)
+ModuleType = TypeVar("ModuleType", bound=nn.Module)
+
+
+@dataclass
+class ModuleFactoryV1(Generic[ConfigType, ModuleType]):
+    """
+    Dataclass for a combination of a Subassembly/Part and the corresponding configuration.
+    Also provides a function to construct the corresponding object through this dataclass
+    """
+
+    module_class: Type[ModuleType]
+    cfg: ConfigType
+
+    def __call__(self) -> ModuleType:
+        """Constructs an instance of the given module class"""
+        return self.module_class(self.cfg)
+
+    def __post_init__(self) -> None:
+        assert signature(self.module_class).parameters["cfg"].annotation == type(self.cfg)
```

Comment:
We shouldn't force the parameter to be called "cfg". Rather check that there is a single parameter and check the type of that one.
```suggestion
        parameters = signature(self.module_class).parameters.values()
        assert len(parameters) == 1
        cfg_parameter = next(iter(parameters))
        assert typeguard.check_type(self.cfg, cfg_parameter.annotation)
```